common/modulecontext/builder.go

```diff
@@ -0,0 +1,98 @@
+package modulecontext
+
+import (
+	"context"
+
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	cf "github.com/TBD54566975/ftl/common/configuration"
+)
+
+type dsnEntry struct {
+	dbType DBType
+	dsn    string
+}
+
+type Builder struct {
+	moduleName string
+	configs    map[string][]byte
+	secrets    map[string][]byte
+	dsns       map[string]dsnEntry
+}
+
+func NewBuilder(moduleName string) *Builder {
+	return &Builder{
+		moduleName: moduleName,
+		configs:    map[string][]byte{},
+		secrets:    map[string][]byte{},
+		dsns:       map[string]dsnEntry{},
+	}
+}
+
+func NewBuilderFromProto(moduleName string, response *ftlv1.ModuleContextResponse) *Builder {
+	configs := map[string][]byte{}
+	for name, bytes := range response.Configs {
+		configs[name] = bytes
+	}
+	secrets := map[string][]byte{}
+	for name, bytes := range response.Secrets {
+		secrets[name] = bytes
+	}
+	dsns := map[string]dsnEntry{}
+	for _, d := range response.Databases {
+		dsns[d.Name] = dsnEntry{dbType: DBType(d.Type), dsn: d.Dsn}
+	}
+	return &Builder{
+		moduleName: moduleName,
+		configs:    configs,
+		secrets:    secrets,
+		dsns:       dsns,
+	}
+}
+
+func (b *Builder) Build(ctx context.Context) (*ModuleContext, error) {
```

Comment:
Given there are now no "builder" methods, I'd just make this a single global function `modulecontext.FromProto() (*ModuleContext, error)` and remove anything Builder related as it seems like an unnecessary abstraction for now. We can revisit once we get around to fleshing out the testing story.

Reply:
moved the builder to the test PR